pkg/segment/aggregations/streamstats.go

```diff
@@ -516,14 +616,73 @@ func PerformWindowStreamStatsOnSingleFunc(currIndex int, ssOption *structs.Strea
 	return latestResult, true, nil
 }
 
-func PerformStreamStatsOnSingleFunc(currIndex int, bucketKey string, ssOption *structs.StreamStatsOptions, measureFuncIndex int, measureAgg *structs.MeasureAggregator, record map[string]interface{}, timestamp uint64, timeSortAsc bool) (interface{}, bool, error) {
+func CreateCValueFromValueExpression(measureAgg *structs.MeasureAggregator, fieldToValue map[string]utils.CValueEnclosure, colValue utils.CValueEnclosure) (utils.CValueEnclosure, bool) {
+	if measureAgg.ValueColRequest == nil {
+		return colValue, true
+	}
+	if measureAgg.ValueColRequest.BooleanExpr != nil {
+		conditionPassed, err := measureAgg.ValueColRequest.BooleanExpr.Evaluate(fieldToValue)
+		if err != nil || !conditionPassed {
+			return utils.CValueEnclosure{}, false
```

Comment:
Do you think there's any case were it would be useful for us to get an error logged when `err != nil` here? If so, you should split this `if` statement and log the error if `err != nil`

Reply:
No, from what I have seen, boolean expression evaluation is totally ignored when the evaluation is incorrect, like performing numeric operations on strings with numbers. Even if we do, it will fill up the log pretty quickly since error would be logged for all the column values.